### PR TITLE
Feature prepareupdate

### LINKED ISSUE #272: prepareUpdate method for pre-parsing SPARQL update statements
i.e. matching prepareQuery

### PR DESCRIPTION
**Pull request by gjhiggins**
Fixes #272
Slightly speculative, other option is close #272 with `wontfix`

## Proposed Changes
Basically, follow the pattern established in `prepareQuery`

  - Create an Update class
  - Add a `prepareUpdate` to `processor.py`
  - Propagate pattern
  - Add test.

### CODE REVIEW

**Review by aucampia — APPROVED** [review]
Looks great.

Notes for other reviewers: I checked that all uses of `rdflib.plugins.sparql.algebra.translateUpdate` will work with new return value, and they will since they all just pass to `rdflib.plugins.sparql.update.evalUpdate`'s second parameter, `update` and this function was updated to access `update.algebra` instead of `update`.

File: rdflib/plugins/sparql/algebra.py [hunk 1]
```diff
@@ -761,7 +761,7 @@ def translateUpdate(q, base=None, initNs=None):
 
         res.append(translateUpdate1(u, prologue))
 
-    return res
+    return Update(prologue, res)
```
Comment (aucampia) [inline rdflib/plugins/sparql/algebra.py:764]: Just a technical note, similar to what I raised in #1626 

This technically changes the RDFLib API, I'm not sure that we consider the methods in here part of the public API, but I think if we don't we should make a note in the docs. Happy to do that if there is agreement on it.

Comment (gjhiggins) [reply rdflib/plugins/sparql/algebra.py:764]: > This technically changes the RDFLib API, I'm not sure that we consider the methods in here part of the public API, but I think if we don't we should make a note in the docs. Happy to do that if there is agreement on it.

I guess from a pragmatic perspective, "public API" is anything that *can* be imported and the SPARQL API is very sparsely documented but yes, agreed.

Comment (aucampia) [reply rdflib/plugins/sparql/algebra.py:764]: I added it to me my to-do list to note these methods are not considered part of the public interface and may break compatibility within a single major version and will do it in a separate PR.

Comment (nicholascar) [reply rdflib/plugins/sparql/algebra.py:764]: Perhaps it's breaking the public API but it's making it more consistent so I suppose we just release a slightly more significant version to include this, perhaps 6.2.0.

**Review by nicholascar — APPROVED** [review]

**Pull request merged**